src/component/legend/LegendView.js

```diff
@@ -361,9 +452,52 @@ export default echarts.extendComponentView({
         );
 
         var contentRect = contentGroup.getBoundingRect();
-        contentGroup.attr('position', [-contentRect.x, -contentRect.y]);
+        var contentPos = [-contentRect.x, -contentRect.y];
+
+        if (selectorGroup) {
+            // Place buttons in selectorGroup
+            selectorGroup && layoutUtil.box(
+                // Buttons in selectorGroup always layout horizontally
+                'horizontal',
+                selectorGroup,
+                legendModel.get('selectorItemGap', true)
+            );
+
+            var selectorRect = selectorGroup.getBoundingRect();
+            var selectorPos = [-selectorRect.x, -selectorRect.y];
+
+            var selectorLegendGap = retrieve2(
+                legendModel.get('selectorLegendGap', true), legendModel.get('itemGap', true)
```

Comment:
I don't think it is a good idea to use legend.itemGap as the default value since they have different appearance.
(but I'm not sure)